### PR TITLE
Todo-mvp clearing issues backlog

### LINKED ISSUE #287: [all branches] Launching activities from Nav Drawer shouldn't clear back stack
Also, add/edit navigation tests.

### PR DESCRIPTION
**Pull request by JoseAlcerreca**
Bumps versions and fixes #287 and #246, completing #301.

### CODE REVIEW

File: todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskActivity.java [hunk 1]
```diff
@@ -39,6 +40,8 @@
 
     private AddEditTaskPresenter mAddEditTaskPresenter;
 
+    private ActionBar mActionBar;
```
Comment (malmstein) [inline todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskActivity.java]: as @ErikHellman points out, we don't need to store the `ActionBar`

File: todoapp/build.gradle [hunk 2]
```diff
@@ -24,12 +24,12 @@ task clean(type: Delete) {
 ext {
     // Sdk and tools
     minSdkVersion = 10
-    targetSdkVersion = 24
-    compileSdkVersion = 24
-    buildToolsVersion = '24.0.2'
+    targetSdkVersion = 25
```
Comment (malmstein) [inline todoapp/build.gradle]: shouldn't we do this as part of a different PR? same with the Support Library Versions

File: todoapp/app/src/androidTest/java/com/example/android/architecture/blueprints/todoapp/tasks/AppNavigationTest.java [hunk 3]
```diff
@@ -113,4 +94,66 @@ public void clickOnAndroidHomeIcon_OpensNavigation() {
...
+        openStatisticsScreen();
+
+        // Open the tasks screen to restore the task
+        openTasksScreen();
+
+        // Pressing back should exit app
+        assertPressingBackExitsApp();
+    }
+
+    private void assertPressingBackExitsApp() {
```
Comment (malmstein) [inline todoapp/app/src/androidTest/java/com/example/android/architecture/blueprints/todoapp/tasks/AppNavigationTest.java]: `assertTrue(activityTestRule.getActivity().isFinishing());` maybe? /cc @ErikHellman

File: todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskActivity.java (hunk 1 above)
Comment (JoseAlcerreca) [inline todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskActivity.java]: We do

File: todoapp/build.gradle (hunk 2 above)
Comment (JoseAlcerreca) [inline todoapp/build.gradle]: Yeah we should

File: todoapp/app/src/androidTest/java/com/example/android/architecture/blueprints/todoapp/tasks/AppNavigationTest.java (hunk 3 above)
Comment (JoseAlcerreca) [inline todoapp/app/src/androidTest/java/com/example/android/architecture/blueprints/todoapp/tasks/AppNavigationTest.java]: Bad idea.
